### DISCUSSION TITLE
[Bug] ollama请求地址错误

### DISCUSSION THREAD

**Discussion by jidaojiuyou** [lobehub/lobehub#3317] [Self Hosting] [answered]
### 📦 Environment

Docker

### 📌 Version

latest

### 💻 Operating System

Windows

### 🌐 Browser

Edge

### 🐛 Bug Description

我用以下命令启动：docker run -d -p 3210:3210 -e OLLAMA_PROXY_URL=http://host.docker.internal:11434/v1 lobehub/lobe-chat
我在页面上使用ollama的qwen2和llama3提问。返回都是
```
{
  "error": {
    "message": "404 page not found",
    "name": "ResponseError",
    "status_code": 404
  },
  "provider": "ollama"
}
```

我在ollama的日志也看到

![image](https://github.com/lobehub/lobe-chat/assets/24694117/8b4a294f-3d40-4db2-9c45-a47ef1d03955)

如图所示前面的三次404都是我发起请求的响应。

后来我在设置里面修改了 `使用客户端请求模式` 为启用。结果就能正常请求并返回了。同时上图第六条就是正常返回时的请求地址。

![image](https://github.com/lobehub/lobe-chat/assets/24694117/1bffea7b-887b-4fde-97f0-9f97a42eea54)
这个就是 `启用` `使用客户端请求模式` 设置发起的请求。
![image](https://github.com/lobehub/lobe-chat/assets/24694117/b1361aa6-0f8d-42f2-9043-1d0a9832ad84)

![image](https://github.com/lobehub/lobe-chat/assets/24694117/c89fa310-2bcf-4ca4-8600-45148966502c)
上面的就是 `未启用` `使用客户端请求模式` 设置发起的请求。

### 📷 Recurrence Steps

0. 设置ollama允许跨域
1. ollama run qwen2 或者 ollama run llama3
2. docker run -d -p 3210:3210 -e OLLAMA_PROXY_URL=http://host.docker.internal:11434/v1 lobehub/lobe-chat
3. 浏览器打开127.0.0.1:3210
4. 选择ollama的模型qwen2或者llama3
5. 提问问题

### 🚦 Expected Behavior

ollama的模型能够正常响应内容，而非404.

[a comment by an automated account is not shown]

**Comment by richardstevenhack** [comment]
Sounds like the same problem I'm having:
Invalid Ollama configuration, please check Ollama configuration and try again #3047

If you do a Google search, it's hard to find anyone who HAS connected Docker to Ollama across any number of similar projects as Lobe Chat. I'm coming to the conclusion that Docker on some Linux distros simply does not work or requires far too much fiddling compared to running a Flatpak or an AppImage.

**Comment by richardstevenhack** [comment, marked as answer]
I just had a similar problem with the Ragapp utility; wouldn't connect to Ollama via Docker. I just solved that problem a few minutes ago by doing this:

Based on an article on the Open-WebUI Github, they suggest:

`If you're experiencing connection issues, it’s often due to the WebUI docker container not being able to reach the Ollama server at 127.0.0.1:11434 (host.docker.internal:11434) inside the container . Use the --network=host flag in your docker command to resolve this. Note that the port changes from 3000 to 8080, resulting in the link: http://localhost:8080/.

Example Docker Command:

docker run -d --network=host -v open-webui:/app/backend/data -e OLLAMA_BASE_URL=http://127.0.0.1:11434/ --name open-webui --restart always ghcr.io/open-webui/open-webui:main

`
What this does for open WebUI according to the Docker documentation is:
1) --network=host set the Docker network to the host network (supposedly this only works on Linux.)
2) Then the Ollama Base URL is set to Ollama's actual host URL.
3) When this is done the port assignments are ignored, so it depends on what port the app is listening on. That appears to work fine with the port 8000 used by the Ragapp. You get a warning message from Docker that the port mappins are ignored.

So I executed this command for Lobe Chat:

```
docker run -d --network=host -p 3210:3210 -e OLLAMA_PROXY_URL=http://127.0.0.1:11434 lobehub/lobe-chat
```

Then I entered the interface at port 3210, selected Ollama as the provider and left the Interface proxy address BLANK. The check passed, I went to the Just Chat, selected qwen2, and entered "Hi" as the message. I got the response from Qwen2.

I don't know if connecting Docker to the local host network is a security risk, but since I'm running on my own machine with the usual firewalls and the like, I don't really care. Your mileage may vary.

I'll also note that I'm running on openSUSE Tumbleweed Linux. For Windows or another Linux,  your mileage may vary.

**Comment by arvinxx** [comment]
@jidaojiuyou 去掉 `/v1`

[a comment by an automated account is not shown]
